### ISSUE TITLE
Reactor 3.1 support

### ISSUE THREAD

**Issue by dbacinski** [JakeWharton/retrofit2-reactor-adapter#4]
In Reactor 3.1 Cancellable interface was replaced by Disposable, and now retrofit2-reactor-adapter is crashing with an error:

```
java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: reactor/core/Cancellation
	at com.jakewharton.retrofit2.adapter.reactor.CallSinkConsumer.accept(CallSinkConsumer.java:35) ~[retrofit2-reactor-adapter-1.0.0.jar:?]
	at com.jakewharton.retrofit2.adapter.reactor.CallSinkConsumer.accept(CallSinkConsumer.java:24) ~[retrofit2-reactor-adapter-1.0.0.jar:?]
	at reactor.core.publisher.FluxCreate.subscribe(FluxCreate.java:90) ~[reactor-core-3.1.0.M1.jar:3.1.0.M1]
```

https://github.com/JakeWharton/retrofit2-reactor-adapter/blob/master/src/main/java/com/jakewharton/retrofit2/adapter/reactor/CallSinkConsumer.java#L35

https://github.com/reactor/reactor-core/releases/tag/v3.1.0.M1

**Comment by dbacinski** [comment]
https://github.com/JakeWharton/retrofit2-reactor-adapter/pull/5

**Comment by JakeWharton** [comment]
So they made a backwards incompatible API change without a major version
bump? Lovely...

**Comment by utwyko** [comment]
> So they made a backwards incompatible API change without a major version
bump? Lovely...

Right, the Project Reactor versioning scheme is a bit weird. This is how they describe it in the [announcement for the Reactor Bismuth M1 release](https://spring.io/blog/2017/05/16/reactor-bismuth-release-train-first-milestone-available#_footnoteref_1):

> As a reminder, Reactor 3 is not entirely conforming to the MAJOR.MINOR.PATCH scheme of semantic versioning, but rather follows a X.MAJOR.MINOR versioning scheme. X incrementing means a major architectural and API do-over, a MAJOR can contain important new features and breaking changes while MINOR are all binary and API compatible within the same MAJOR (although they can contain internals changes and new features in addition to bugfixes)

**Comment by colesnodgrass** [comment]
https://github.com/reactor/reactor-core/releases/tag/v3.1.0.RELEASE 

3.1.0 has been released, any chance of support being added soon?

**Comment by JakeWharton** [comment]
Looking into it this week.

**Comment by zlangbert** [comment]
@JakeWharton Are you planning a release anytime soon? Would be great to start using this.

**Comment by JakeWharton** [comment]
Yes. Will try it now.

**Comment by JakeWharton** [comment]
Maven release plugin sucks. Porting to Gradle tonight or tomorrow and I'll
get it out.

**Comment by JakeWharton** [comment]
Didn't get to this today. Will tomorrow for sure.

**Comment by JakeWharton** [comment]
Got 90% of the way to Gradle conversion. Should be able to knock this out
this weekend or early next week.

**Comment by JakeWharton** [comment]
Released as 2.0. Thanks for your patience!

**Comment by zlangbert** [comment]
Awesome, thank you!
